lib/spack/llnl/util/filesystem.py

```diff
@@ -1180,6 +1179,46 @@ def remove_if_dead_link(path):
         os.unlink(path)
 
 
+class GenericVisitor(object):
+    def __init__(self, callable):
+        pass
```

Comment:
```suggestion
```

Reply:
I agree with this suggestion (I don't think the `__init__` in the interface is useful for enforcing policy in subclasses)